src/main/java/com/google/cloud/discotoproto3converter/proto3/DocumentToProtoConverter.java

```diff
@@ -306,37 +305,36 @@ private void readResources(Document document) {
         String inputDescription = getInputMessageDescription(grpcServiceName, methodname);
         Message input = new Message(requestName, false, false, inputDescription);
         String httpOptionPath = method.flatPath();
-        List<String> methodSignatureParams = new ArrayList<>();
-
-        for (Schema requiredParam : method.requiredParams().values()) {
-          Field requiredField = schemaToField(requiredParam);
-          Option fieldBehaviorOption = new Option("google.api.field_behavior");
-          fieldBehaviorOption.getProperties().put("", "REQUIRED");
-          requiredField.getOptions().add(fieldBehaviorOption);
-          input.getFields().add(requiredField);
-          methodSignatureParams.add(requiredField.getName());
+        Map<String, String> methodSignatureParamNames = new LinkedHashMap<>();
+        for (String requiredParamName : method.requiredParamNames()) {
+          methodSignatureParamNames.put(requiredParamName, null);
         }
 
         for (Schema pathParam : method.pathParams().values()) {
           Field pathField = schemaToField(pathParam);
+          if (methodSignatureParamNames.containsKey(pathParam.getIdentifier())) {
+            pathField.getOptions().add(getFieldBehaviorOption("REQUIRED"));
```

Comment:
(minor for now) This `"REQUIRED"` should be an enum, and the function does what it needs to with it.

Reply:
It should be, probably, but there are many things like this in this whole project. I think keeping it less pretty but consistent with each other is better than keeping one thing pretty but inconsistent.

Reply:
Looking into this more, I still believe that not having an enum is better because this method is private. I.e. the whole option creation thing is not exposed anywhere beyond this class, and thus using string directly becomes the class implementation detail. If we put that value in enum, I don't even know how to name that enum - it is a field-behavior enum, which can have only one value as of now.